### ISSUE TITLE
Compilation errors with MinGW 5.3 + MSYS

### ISSUE THREAD

**Issue by HighCommander4** [google/googletest#1060]
I'm trying to build a project that uses googletest ([cpp-netlib](https://github.com/cpp-netlib/cpp-netlib)) with MinGW 5.3 and MSYS, but I'm running into a bunch of compiler errors (shown at the bottom).

I reported an [issue against the downstream project](https://github.com/cpp-netlib/cpp-netlib/issues/745), and they suggested I report one here.

I'm using the 32-bit version of MinGW from MinGW.org (and the corresponding MSYS), but another user [reports](https://github.com/cpp-netlib/cpp-netlib/issues/745#issuecomment-290784461) seeing the same thing with MinGW64 and MSYS2.

Here are the errors:

```
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: In funct
ion 'void testing::internal::SleepMilliseconds(int)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1453:24:
error: 'nanosleep' was not declared in this scope
   nanosleep(&time, NULL);
                        ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: At globa
l scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1782:3: e
rror: 'AutoHandle' does not name a type
   AutoHandle thread_;
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: In funct
ion 'int testing::internal::posix::StrCaseCmp(const char*, const char*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:2318:25:
error: '_stricmp' was not declared in this scope
   return _stricmp(s1, s2);
                         ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: In funct
ion 'char* testing::internal::posix::StrDup(const char*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:2320:58:
error: '_strdup' was not declared in this scope
 inline char* StrDup(const char* src) { return _strdup(src); }
                                                          ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: In funct
ion 'int testing::internal::posix::FileNo(FILE*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:2328:52:
error: '_fileno' was not declared in this scope
 inline int FileNo(FILE* file) { return _fileno(file); }
                                                    ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h: In funct
ion 'FILE* testing::internal::posix::FDOpen(int, const char*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:2374:71:
error: 'fdopen' was not declared in this scope
 inline FILE* FDOpen(int fd, const char* mode) { return fdopen(fd, mode); }
                                                                       ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:42:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest.cc: In static member function 'stat
ic bool testing::internal::String::CaseInsensitiveWideCStringEquals(const wchar_t*, const wchar_t*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest.cc:1932:27: error: '_wcsicmp' was n
ot declared in this scope
   return _wcsicmp(lhs, rhs) == 0;
                           ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:43:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: In function 'void te
sting::internal::DeathTestAbort(const string&)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:273:12: error: '_exit
' was not declared in this scope
     _exit(1);
            ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: In member function '
virtual void testing::internal::DeathTestImpl::Abort(testing::internal::DeathTest::AbortReason)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:498:10: error: '_exit
' was not declared in this scope
   _exit(1);  // Exits w/o any normal exit hooks (we were supposed to crash)
          ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: At global scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:637:3: error: 'AutoHa
ndle' does not name a type
   AutoHandle write_handle_;
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:639:3: error: 'AutoHa
ndle' does not name a type
   AutoHandle child_handle_;
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:644:3: error: 'AutoHa
ndle' does not name a type
   AutoHandle event_handle_;
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: In member function '
virtual int testing::internal::WindowsDeathTest::Wait()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:656:36: error: 'child
_handle_' was not declared in this scope
   const HANDLE wait_handles[2] = { child_handle_.Get(), event_handle_.Get() };
                                    ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:656:57: error: 'event
_handle_' was not declared in this scope
   const HANDLE wait_handles[2] = { child_handle_.Get(), event_handle_.Get() };
                                                         ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:670:3: error: 'write_
handle_' was not declared in this scope
   write_handle_.Reset();
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: In member function '
virtual testing::internal::DeathTest::TestRole testing::internal::WindowsDeathTest::AssumeRole()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:720:3: error: 'write_
handle_' was not declared in this scope
   write_handle_.Reset(write_handle);
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:721:3: error: 'event_
handle_' was not declared in this scope
   event_handle_.Reset(::CreateEvent(
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:741:24: error: '_MAX_
PATH' was not declared in this scope
   char executable_path[_MAX_PATH + 1];  // NOLINT
                        ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:744:45: error: 'execu
table_path' was not declared in this scope
                                             executable_path,
                                             ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:285:38: note: in defi
nition of macro 'GTEST_DEATH_TEST_CHECK_'
     if (!::testing::internal::IsTrue(expression)) { \
                                      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:767:7: error: 'execut
able_path' was not declared in this scope
       executable_path,
       ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:285:38: note: in defi
nition of macro 'GTEST_DEATH_TEST_CHECK_'
     if (!::testing::internal::IsTrue(expression)) { \
                                      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:777:3: error: 'child_
handle_' was not declared in this scope
   child_handle_.Reset(process_info.hProcess);
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc: In function 'int tes
ting::internal::GetStatusFileDescriptor(unsigned int, size_t, size_t)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:1230:3: error: 'AutoH
andle' was not declared in this scope
   AutoHandle parent_process_handle(::OpenProcess(PROCESS_DUP_HANDLE,
   ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:1233:7: error: 'paren
t_process_handle' was not declared in this scope
   if (parent_process_handle.Get() == INVALID_HANDLE_VALUE) {
       ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:1249:26: error: 'pare
nt_process_handle' was not declared in this scope
   if (!::DuplicateHandle(parent_process_handle.Get(), write_handle,
                          ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-death-test.cc:1264:26: error: 'pare
nt_process_handle' was not declared in this scope
   if (!::DuplicateHandle(parent_process_handle.Get(), event_handle,
                          ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:44:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-filepath.cc: In static member funct
ion 'static testing::internal::FilePath testing::internal::FilePath::GetCurrentDir()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-filepath.cc:52:26: error: '_MAX_PAT
H' was not declared in this scope
 # define GTEST_PATH_MAX_ _MAX_PATH
                          ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-filepath.cc:105:12: note: in expans
ion of macro 'GTEST_PATH_MAX_'
   char cwd[GTEST_PATH_MAX_ + 1] = { '\0' };
            ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-filepath.cc:106:27: error: 'cwd' wa
s not declared in this scope
   return FilePath(_getcwd(cwd, sizeof(cwd)) == NULL ? "" : cwd);
                           ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In function 'void testing:
:internal::SleepMilliseconds(int)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:179:6: error: redefinition
of 'void testing::internal::SleepMilliseconds(int)'
 void SleepMilliseconds(int n) {
      ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1448:13:
note: 'void testing::internal::SleepMilliseconds(int)' previously defined here
 inline void SleepMilliseconds(int n) {
             ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: At global scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:183:1: error: 'AutoHandle'
does not name a type
 AutoHandle::AutoHandle()
 ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:186:1: error: 'AutoHandle'
does not name a type
 AutoHandle::AutoHandle(Handle handle)
 ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:189:1: error: 'AutoHandle'
does not name a type
 AutoHandle::~AutoHandle() {
 ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:193:1: error: 'AutoHandle'
does not name a type
 AutoHandle::Handle AutoHandle::Get() const {
 ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:197:6: error: 'AutoHandle'
has not been declared
 void AutoHandle::Reset() {
      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In function 'void testing:
:internal::Reset()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:198:29: error: too many arg
uments to function 'void testing::internal::Reset()'
   Reset(INVALID_HANDLE_VALUE);
                             ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:197:6: note: declared here
 void AutoHandle::Reset() {
      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: At global scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:201:6: error: 'AutoHandle'
has not been declared
 void AutoHandle::Reset(HANDLE handle) {
      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In function 'void testing:
:internal::Reset(HANDLE)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:203:7: error: 'handle_' was
 not declared in this scope
   if (handle_ != handle) {
       ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:204:21: error: 'IsCloseable
' was not declared in this scope
     if (IsCloseable()) {
                     ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:209:31: error: 'IsCloseable
' was not declared in this scope
     GTEST_CHECK_(!IsCloseable())
                               ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1297:37:
note: in definition of macro 'GTEST_CHECK_'
     if (::testing::internal::IsTrue(condition)) \
                                     ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: At global scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:215:6: error: 'AutoHandle'
has not been declared
 bool AutoHandle::IsCloseable() const {
      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:215:32: error: non-member f
unction 'bool testing::internal::IsCloseable()' cannot have cv-qualifier
 bool AutoHandle::IsCloseable() const {
                                ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In function 'bool testing:
:internal::IsCloseable()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:218:10: error: 'handle_' wa
s not declared in this scope
   return handle_ != NULL && handle_ != INVALID_HANDLE_VALUE;
          ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: At global scope:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:221:1: error: redefinition
of 'testing::internal::Notification::Notification()'
 Notification::Notification()
 ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1470:3: n
ote: 'testing::internal::Notification::Notification()' previously defined here
   Notification() : notified_(false) {
   ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:229:6: error: redefinition
of 'void testing::internal::Notification::Notify()'
 void Notification::Notify() {
      ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1479:8: n
ote: 'void testing::internal::Notification::Notify()' previously defined here
   void Notify() {
        ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:233:6: error: redefinition
of 'void testing::internal::Notification::WaitForNotification()'
 void Notification::WaitForNotification() {
      ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1487:8: n
ote: 'void testing::internal::Notification::WaitForNotification()' previously defined here
   void WaitForNotification() {
        ^
In file included from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:45:0:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In constructor 'testing::i
nternal::Mutex::Mutex()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:242:45: error: cannot conve
rt 'CRITICAL_SECTION* {aka _CRITICAL_SECTION*}' to '_RTL_CRITICAL_SECTION*' in initialization
       critical_section_(new CRITICAL_SECTION) {
                                             ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:243:48: error: cannot conve
rt '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void InitializeCritica
lSection(LPCRITICAL_SECTION)'
   ::InitializeCriticalSection(critical_section_);
                                                ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In destructor 'testing::in
ternal::Mutex::~Mutex()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:253:46: error: cannot conve
rt '_RTL_CRITICAL_SECTION*' to 'PCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void DeleteCriticalSect
ion(PCRITICAL_SECTION)'
     ::DeleteCriticalSection(critical_section_);
                                              ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In member function 'void t
esting::internal::Mutex::Lock()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:261:43: error: cannot conve
rt '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void EnterCriticalSect
ion(LPCRITICAL_SECTION)'
   ::EnterCriticalSection(critical_section_);
                                           ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In member function 'void t
esting::internal::Mutex::Unlock()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:271:43: error: cannot conve
rt '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void LeaveCriticalSect
ion(LPCRITICAL_SECTION)'
   ::LeaveCriticalSection(critical_section_);
                                           ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In member function 'void t
esting::internal::Mutex::ThreadSafeLazyInit()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:292:27: error: cannot conve
rt 'CRITICAL_SECTION* {aka _CRITICAL_SECTION*}' to '_RTL_CRITICAL_SECTION*' in assignment
         critical_section_ = new CRITICAL_SECTION;
                           ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:293:54: error: cannot conve
rt '_RTL_CRITICAL_SECTION*' to 'LPCRITICAL_SECTION {aka _CRITICAL_SECTION*}' for argument '1' to 'void InitializeCritica
lSection(LPCRITICAL_SECTION)'
         ::InitializeCriticalSection(critical_section_);
                                                      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In constructor 'testing::i
nternal::ThreadWithParamBase::ThreadWithParamBase(testing::internal::ThreadWithParamBase::Runnable*, testing::internal::
Notification*)':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:377:9: error: class 'testin
g::internal::ThreadWithParamBase' does not have any field named 'thread_'
       : thread_(ThreadWithParamSupport::CreateThread(runnable,
         ^
In file included from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/
gtest-internal.h:40:0,
                 from C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/gtest.h:5
8,
                 from c:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-all.cc:39:
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc: In member function 'void t
esting::internal::ThreadWithParamBase::Join()':
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/src/gtest-port.cc:386:38: error: 'thread_' wa
s not declared in this scope
   GTEST_CHECK_(::WaitForSingleObject(thread_.Get(), INFINITE) == WAIT_OBJECT_0)
                                      ^
C:/Users/nr/dev/libraries/cpp-netlib-0.12.0/deps/googletest/googletest/include/gtest/internal/gtest-port.h:1297:37:
note: in definition of macro 'GTEST_CHECK_'
     if (::testing::internal::IsTrue(condition)) \
                                     ^
```

**Comment by Marc--Olivier** [comment]
Seems similar to #1060

**Comment by Marc--Olivier** [comment]
Does passing -Dgtest_disable_pthreads=ON to CMake fix the problem? (see #975)

**Comment by gennadiycivil** [comment]
I have been cleaning up older and inactive GitHub googletest issues. You may see an issue "closed" if it appears to be inactive/abandoned 
Thank you
